gateway/ln-gateway/src/lib.rs

```diff
@@ -131,6 +131,11 @@ impl LnGateway {
             LnGatewayError::Other(anyhow::anyhow!("Invalid federation member string {}", e))
         })?;
 
+        // The gateway deterministically assigns a channel id (u64) to each federation connected.
+        // At present, the maximum number of federations that can be referenced in the actors map
+        // is equivalent to `usize.max_value()`,
+        let new_channel_id: u64 = self.actors.lock().await.len() as u64;
```

Comment:
I think generating a random ID might be better. 

For instance, federation A is joined, federation B is joined, federation A is left, federation C is joined. Federation A & C would have the same channel id, no?

Reply:
Good point, there's no guarantee the number of actors is always increasing and we may have channel collisions because of my current assumption here. However, picking the channel ID at random reduces efficiency of utilizing the `u64` set of channels possible. Also, we'd need to keep track of what random ID is already assigned (looks like that's whats missing). 

We should keep a registry of what channel id is already assigned, or freed after previous assignment. In addition to picking the channel id deterministically (of which I propose picking the next available ID in u64 number sequence)

On gateway reload, we simply assign IDs in sequence as we have implemented already

Reply:
marked with a todo to follow up on shortly.

Pending, but should not block this PR it think

Reply:
Included fix: Keep track of previously assigned channels to guarantee unique assignments. Do this optimally by utilizing the entire `u64` set, starting from channel id `0` to `usize.max_value()` <- likely never gets to `u64` channels.
```
async fn assign_channel_id(&self) -> u64 {
    // The gateway deterministically assigns a channel id (u64) to each federation connected.
    // At present, the maximum number of federations that can be referenced in the actors map
    // is equivalent to `usize.max_value()`,

    let mut channels = self.channels.lock().await;
    let channels_n = channels.len() as u64;

    let actors_n = self.actors.lock().await.len() as u64;

    if actors_n == channels_n {
        channels.insert(actors_n);
        actors_n
    } else {
        let mut channel_id = 0;
        while channels.contains(&channel_id) {
            channel_id += 1;
        }
        channels.insert(channel_id);
        channel_id
    }
}
```

Reply:
Latest iteration uses `AtomicU64` to select and assign next id. Check it out